R/dashboard_modules/04-learner_characteristics.R

```diff
@@ -1,5 +1,253 @@
+  ) %>%
+  mutate(measure = firstup(measure))
+
+
+
+
+# Create static lists of options for dropdowns
+chars_year_choices <- sort(data_choices(data = chars_parquet, column = "year"),
+  decreasing = TRUE
+)
+chars_measure_choices <- data_choices(data = chars_parquet, column = "measure")
+# for providers need to remove total first so can put at the beginning
+chars_parquet_no_total_providers <- chars_parquet %>%
+  filter(provider_name != "Total")
+chars_provider_choices <- sort(data_choices(
+  data = chars_parquet_no_total_providers,
+  column = "provider_name"
+))
+
+
+
+# Main module code ============================================================
+
+learner_characteristics_ui <- function(id) {
   div(
+    # Tab header ==============================================================
     h1("Learner characteristics"),
+
+    # User selection area =====================================================
+    div(
+      class = "well",
+      style = "min-height: 100%; height: 100%; overflow-y: visible;",
+      bslib::layout_column_wrap(
+        width = "15rem", # Minimum width for each input box before wrapping
+        selectizeInput(
+          inputId = NS(id, "provider"),
+          label = NULL,
+          choices = NULL,
+          selected = "Total",
+          options = list(maxOptions = 2000)
+        ),
+        selectInput(
+          inputId = NS(id, "year"),
+          label = "Select academic year",
+          choices = c(chars_year_choices),
+          selected = "2023/24 (Q3 Aug to Apr)"
+        ),
+        selectInput(
+          inputId = NS(id, "measure"),
+          label = "Select measure",
+          choices = c(chars_measure_choices),
+          selected = "Starts"
+        ),
+      )
+    ),
+
+    # Main table ==============================================================
+    navset_card_tab(
+      id = "provider_learner_characteristics",
+      ## lldd tab ------------------------------------------------------------
+      nav_panel(
+        "LLDD",
+        plotOutput(NS(id, ("lldd_plot")))
+      ),
+      ## sex tab ------------------------------------------------------------
+      nav_panel(
+        "Sex",
+        plotOutput(NS(id, ("sex_plot")))
+      ),
+      ## Age tab ------------------------------------------------------------
+      nav_panel(
+        "Age",
+        plotOutput(NS(id, ("age_plot")))
+      ),
+      # Ethnicity tab ------------------------------------------------------------
+      nav_panel(
+        "Ethnicity",
+        plotOutput(NS(id, ("ethnicity_plot")))
+      ),
+      ## Download tab ---------------------------------------------------------
+      nav_panel(
+        "Download data",
+        shinyGovstyle::radio_button_Input(
+          inputId = NS(id, "file_type"),
+          label = h2("Choose download file format"),
+          hint_label = "The XLSX format is designed for use in Microsoft Excel",
+          choices = c("CSV (Up to 5.22 MB)", "XLSX (Up to 1.76 MB)"),
```

Comment:
Not the most important if the data itself may change, though looks like from the code you added into the check_file_sizes.R file these should be 8.73 MB for CSV and 2.74 MB for XLSX

![image](https://github.com/user-attachments/assets/7967c303-6856-4b87-9207-e6d2a2b6c2c1)